src/opm/input/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp

```diff
@@ -395,6 +426,19 @@ namespace {
                                       table.getKrwColumn(), tolcrit);
     }
 
+    /// Maximum water saturation for which Krgw(1-Sw) <= tolcrit.
+    ///
+    /// Expected Table Format:
+    ///    [1-Sw,  Krgw(1-Sw), ...other...]
+    ///
+    ///    Krgw decreasing.
+    template <typename T>
+    double critical_water_sgwfn(const T& table, const double tolcrit)
```

Comment:
This should not be a template.  It should take exactly an `SwgfnTable` as its first argument.